File: Serene/Serene.Script/Northwind/Category/CategoryGrid.cs

namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;
    using System;
    using System.Collections.Generic;
    using System.Runtime.CompilerServices;

    [Imported]
    [ColumnsKey("Northwind.Category"), IdProperty("CategoryID"), NameProperty("CategoryName")]
    [DialogType(typeof(CategoryDialog)), LocalTextPrefix("Northwind.Category"), Service("Northwind/Category")]
    public class CategoryGrid : EntityGrid<CategoryRow>
    {
        public CategoryGrid(jQueryObject container)
            : base(container)
        {
        }
    }
}

File: Serene/Serene.Core/Modules/Northwind/Category/CategoryGrid.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    export class CategoryGrid extends Serenity.EntityGrid<CategoryRow, any> {
        protected getColumnsKey() { return "Northwind.Category"; }
        protected getDialogType() { return <any>CategoryDialog; }
        protected getIdProperty() { return CategoryRow.idProperty; }
        protected getLocalTextPrefix() { return CategoryRow.localTextPrefix; }
        protected getService() { return CategoryService.baseUrl; }

        constructor(container: JQuery) {
            super(container);
        }
    }
}